### DISCUSSION TITLE
Debugging with IntelliJ Rust Plugin

### DISCUSSION THREAD

**Discussion by jcsho** [cucumber-rs/cucumber#203] [Q&A] [answered]
Hello all,

Firstly, really appreciate all the work being done here and love the BDD style of testing.

Just a question regarding debugging on IntelliJ platforms using their Rust plugin if anyone has experienced it.

When running a debug session, the plugin seems to inject the arguments `--format=json -Z unstable-options --show-output` and results in failure message below:

```sh
# Clion debug console
~/ray-tracer/target/debug/deps/canvas-4ae8ce55a3139d21 --format=json -vv -Z unstable-options --show-output
error: Found argument '--format' which wasn't expected, or isn't valid in this context

	If you tried to supply `--format` as a value rather than a flag, use `-- --format`

USAGE:
    canvas-4ae8ce55a3139d21 [OPTIONS]

For more information try --help

Process finished with exit code 2
```

I'm unable to remove them even if I pass in extra args like `-vv` in the run configuration

<img width="1152" alt="Screen Shot 2022-02-05 at 12 05 52 AM" src="https://user-images.githubusercontent.com/26936838/152633934-4b3c80aa-b830-4156-a3ec-d6ff4ebf3d73.png">

Currently I am working around it by using rust-lldb for debugging but would like to know if anyone's has a working setup?

**Comment by ilslv** [comment, marked as answer]
@jcsho we are aware of this issue https://github.com/cucumber-rs/cucumber/issues/194 and current solution is to implement custom CLI which will ignore options automatically passed by CLion. See https://github.com/cucumber-rs/cucumber/issues/194#issuecomment-1002884253 and book [chapter on CLI](https://cucumber-rs.github.io/cucumber/main/cli.html#customizing) for more info.

**Reply by jcsho** [reply]
thanks @ilslv!
